src/payment/strategies/googlepay/googlepay-payment-processor.ts

```diff
@@ -180,14 +165,40 @@ export default class GooglePayPaymentProcessor {
         return new google.payments.api.PaymentsClient({ environment });
     }
 
-    private _mapGooglePayAddressToBillingAddress(address: GooglePayAddress, id: string): BillingAddressUpdateRequestBody {
+    private _getMethodId(): string {
+        if (!this._methodId) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        return this._methodId;
+    }
+
+    private _mapGooglePayAddressToBillingAddress(paymentData: GooglePaymentData, id: string): BillingAddressUpdateRequestBody {
         return {
             id,
+            firstName: paymentData.paymentMethodData.info.billingAddress.name.split(' ').slice(0, -1).join(' '),
+            lastName: paymentData.paymentMethodData.info.billingAddress.name.split(' ').slice(-1).join(' '),
+            company: paymentData.paymentMethodData.info.billingAddress.companyName,
+            address1: paymentData.paymentMethodData.info.billingAddress.address1,
+            address2: paymentData.paymentMethodData.info.billingAddress.address2 + paymentData.paymentMethodData.info.billingAddress.address3,
+            city: paymentData.paymentMethodData.info.billingAddress.locality,
+            stateOrProvince: paymentData.paymentMethodData.info.billingAddress.administrativeArea,
+            stateOrProvinceCode: paymentData.paymentMethodData.info.billingAddress.administrativeArea,
+            postalCode: paymentData.paymentMethodData.info.billingAddress.postalCode,
+            countryCode: paymentData.paymentMethodData.info.billingAddress.countryCode,
+            phone: paymentData.paymentMethodData.info.billingAddress.phoneNumber,
+            customFields: [],
```

Comment:
I notice that custom fields are hardcoded as an empty array. What if the merchant has enabled custom form fields? Are we still passing the values through?

Reply:
There are no custom fields coming back from google pay wallet [Reference](https://developers.google.com/pay/api/web/reference/object#Address)

Reply:
No, custom fields are what BigCommerce offers, not GooglePay. I think if merchants have custom fields enabled and mark them as required, I don't think shoppers are able to check out using Google Pay, because currently, they are not allowed to edit their billing details. Anyway, you can just leave it like this for now and we can address this issue later.